src/component/addons/AddonForm/AddonMultiSelector/AddonMultiSelector.tsx

```diff
@@ -0,0 +1,182 @@
+
+export interface IAddonMultiSelectorProps {
+    options: IAutocompleteBoxOption[];
+    selectedItems: string[];
+    onChange: (value: string[]) => void;
+    error?: string;
+    onFocus?: () => void;
+    entityName: string;
+    selectAllEnabled: boolean;
+}
+
+const ALL_OPTIONS = '*';
+
+export const AddonMultiSelector: VFC<IAddonMultiSelectorProps> = ({
+    options,
+    selectedItems,
+    onChange,
+    error,
+    onFocus,
+    entityName,
+    selectAllEnabled = true,
+}) => {
+    const [isWildcardSelected, selectWildcard] = useState(
+        selectedItems.includes(ALL_OPTIONS)
+    );
+    const renderInput = (params: AutocompleteRenderInputParams) => (
+        <TextField
+            {...params}
+            error={Boolean(error)}
+            helperText={error}
+            variant="outlined"
+            label={`${capitalize(entityName)}s`}
+            placeholder={`Select ${entityName}s to filter by`}
+            onFocus={onFocus}
+            data-testid={`select-${entityName}-input`}
+        />
+    );
+
+    const StyledCheckbox = styled(Checkbox)(() => ({
+        marginRight: '0.2em',
+    }));
+    const isAllSelected =
+        selectedItems.length > 0 &&
+        selectedItems.length === options.length &&
+        selectedItems[0] !== ALL_OPTIONS;
+
+    const onAllItemsChange = (
+        e: ChangeEvent<HTMLInputElement>,
+        checked: boolean
+    ) => {
+        if (checked) {
+            selectWildcard(true);
+            onChange([ALL_OPTIONS]);
+        } else {
+            selectWildcard(false);
+            onChange(selectedItems.includes(ALL_OPTIONS) ? [] : selectedItems);
+        }
+    };
+
+    const onSelectAllClick = () => {
+        const newItems = isAllSelected ? [] : options.map(({ value }) => value);
+        onChange(newItems);
+    };
+    const renderOption = (
+        props: object,
+        option: IAutocompleteBoxOption,
+        { selected }: AutocompleteRenderOptionState
+    ) => {
+        return (
+            <li {...props}>
+                <StyledCheckbox
+                    icon={<CheckBoxOutlineBlankIcon fontSize="small" />}
+                    checkedIcon={<CheckBoxIcon fontSize="small" />}
+                    checked={selected}
+                />
+                {option.label}
+            </li>
+        );
+    };
+    const renderGroup = ({ key, children }: AutocompleteRenderGroupParams) => (
+        <Fragment key={key}>
+            <ConditionallyRender
+                condition={options.length > 2 && selectAllEnabled}
+                show={
+                    <SelectAllButton
+                        isAllSelected={isAllSelected}
+                        onClick={onSelectAllClick}
+                    />
+                }
+            />
+            {children}
+        </Fragment>
+    );
+    const CustomPaper = ({ ...props }) => <Paper elevation={8} {...props} />;
```

Comment:
We should avoid creating react components inside of react components, this pattern may lead to unexpected bugs, especially with dynamic content like input fields and checkboxes.